src/vs/platform/terminal/node/terminalProfiles.ts

```diff
@@ -327,6 +328,7 @@ function applyConfigProfilesToMap(configProfiles: { [key: string]: IUnresolvedTe
 		if (value === null || (!('path' in value) && !('source' in value))) {
 			profilesMap.delete(profileName);
 		} else {
+			value.icon = value.icon || profilesMap.get(profileName)?.icon;
```

Comment:
these configured icons were getting lost/overridden because value.icon is undefined unless provided by the user